### PR TITLE
add solution for day 1

### PR DESCRIPTION
**Pull request by shraddhaag**
This PR adds solution to AoC 2023 day 1. 

To run the solution, navigate to the `day1` dirctory, and run 

```
go run main.go <PATH_TO_YOUR_INPUT_FILE>
```

Looking forward to productive discussions!

### CODE REVIEW

File: day1/main.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+	var first, last int
+
+	if len(charSlice) != 0 && charSlice[0] < intSlice[0] {
+		first = charMap[charSlice[0]]
+	} else {
+		first = int(input[intSlice[0]]) - 48
+	}
+
+	if len(charSlice) != 0 && charSlice[len(charSlice)-1] > intSlice[len(intSlice)-1] {
```
Comment (sugamadhiakri) [inline day1/main.go:88]: Getting `panic: runtime error: index out of range [0] with length 0` in this line.

File: day1/main.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+			charSlice = append(charSlice, last)
+			charMap[last] = intDigit
+		}
+	}
+
+	sort.Ints(charSlice)
+
+	var first, last int
+
+	if len(charSlice) != 0 && charSlice[0] < intSlice[0] {
```
Comment (sugamadhiakri) [inline day1/main.go]: Getting `panic: runtime error: index out of range [0] with length 0` in this line.

Comment (sugamadhiakri) [reply day1/main.go]: The error seems to be coming from empty `intSlice` which happens when there are no actual digits in the line. for example: sixtythree.

Comment (shraddhaag) [reply day1/main.go]: surprisingly I didn't encounter it in my test input. thanks for calling this out. 

I have addressed it though, hope that helps!

**Review by MatthewJamesBoyle — APPROVED** [review]

**Pull request merged**